service_taxi/src/main/kotlin/org/thechance/service_taxi/api/endpoints/TripRoutes.kt

```diff
@@ -87,32 +87,43 @@ fun Route.tripRoutes() {
             socketHandler.collectTripStatus(tripId)
         }
 
+        webSocket("/track/rides/{userId}") {
+            val userId = call.parameters["userId"]?.trim().orEmpty()
+            socketHandler.trips[userId] = WebSocketTrip(session = this)
+            socketHandler.collectTripStatus(userId)
+        }
+
         put("/approve") {
             val parameters = call.receiveParameters()
             val tripId = parameters["tripId"] ?: throw MissingParameterException()
             val driverId = parameters["driverId"] ?: throw MissingParameterException()
             val taxiId = parameters["taxiId"] ?: throw MissingParameterException()
+            val userId = parameters["userId"] ?: throw MissingParameterException()
             val result = driverTripsManagementUseCase.approveTrip(driverId, taxiId, tripId).toDto()
             socketHandler.trips[tripId]?.trip?.emit(result)
-            socketHandler.endSession(driverId) // end collecting from other clients
+            socketHandler.trips[userId]?.trip?.emit(result)
             call.respond(HttpStatusCode.OK, result)
         }
 
         put("/received") {
             val parameters = call.receiveParameters()
             val tripId = parameters["tripId"] ?: throw MissingParameterException()
             val driverId = parameters["driverId"] ?: throw MissingParameterException()
+            val userId = parameters["userId"] ?: throw MissingParameterException()
```

Comment:
Why is the userId parameter required, given that the user creating the trip is already known? Additionally, relying on the delivery mechanism to send the correct userId is not secure.